### PR TITLE
Add Startup probe to charts.

### PR DESCRIPTION
**Pull request by KevinJoiner**
Issue: https://github.com/rancher/rancher/issues/40824

Added the startup probe to the rancher-webhook chart so that the Webhook does not appear to be ready before it has updated the validating webhook configuration.
The startup probe checks every 5 seconds for 10 minutes before declaring that the webhook failed to start.
These values seemed reasonable to me but are up for discussion. `5s periodSeconds * 120 failureThreshold`

In the second commit, I updated the integration test to install the newly packaged charts created in CI to provide us with a larger test coverage. I also added a command that will start tailing the rancher logs if Rancher-Webhook fails to deploy after 5 minutes.

### CODE REVIEW

**Review by rmweir — CHANGES_REQUESTED** [review]
10 minutes seems excessive to me. I think 5 should be enough. If that doesn't sound right to you and other reviewers though then 10 is fine. It's difficult to reason about what to set it to apart from limited experience watching it.

**Comment by KevinJoiner** [comment]
It's probably also worth adding a liveness probe to the deployment with the same endpoint.

**Review by rmweir — APPROVED** [review]

**Review by MbolotSuse — COMMENTED** [review]
A couple of small nits and one minor concern, but I think the overall approach is good.

File: charts/rancher-webhook/templates/deployment.yaml [hunk 1]
```diff
@@ -48,6 +48,21 @@ spec:
...
         - name: capi-https
           containerPort: 8777
+        startupProbe:
+          httpGet:
+            path: "/healthz"
+            port: "https"
+            scheme: "HTTPS"
+          failureThreshold: 60
+          periodSeconds: 5
+          successThreshold: 1
```
[the remark this answers is not in the document]
Comment (KevinJoiner) [reply charts/rancher-webhook/templates/deployment.yaml]: Updated.

File: charts/rancher-webhook/templates/deployment.yaml [hunk 2]
```diff
@@ -48,6 +48,21 @@ spec:
...
+          timeoutSeconds: 1
```
[the remark this answers is not in the document]
Comment (KevinJoiner) [reply charts/rancher-webhook/templates/deployment.yaml]: Updated.

File: scripts/integration-test [hunk 3]
```diff
@@ -6,49 +6,49 @@ export CATTLE_SERVER_URL="https://$(ip route get 8.8.8.8 | awk '{print $7}'):443
...
-            \"timeoutSeconds\": 1
-        }
-    }]}}}}"
+WEBHOOK_REPO=$(ctr image import ${IMAGE_FILE} | cut -d ' ' -f 2 | cut -d ':' -f 1)
+
+# Source tags file to get the last built tags
+source ./dist/tags
+
+# Install the webhook chart we just built.
+helm upgrade rancher-webhook ./dist/artifacts/rancher-webhook-${HELM_VERSION}.tgz -n cattle-system --set image.repository=${WEBHOOK_REPO} --set image.tag=${TAG}
```
[the remark this answers is not in the document]
Comment (KevinJoiner) [reply scripts/integration-test]: Updated.

**Review by MbolotSuse — APPROVED** [review]

**Review by rmweir — APPROVED** [review]

**Pull request merged**